src/patroni.py

```diff
@@ -109,6 +110,25 @@ def are_all_members_ready(self) -> bool:
 
         return all(member["state"] == "running" for member in r.json()["members"])
 
+    @property
+    def primary_endpoint_ready(self) -> bool:
+        """Is the primary endpoint redirecting connections to the primary pod.
+
+        Returns:
+            Return whether the primary endpoint is redirecting connections to the primary pod.
+        """
+        try:
+            for attempt in Retrying(stop=stop_after_delay(10), wait=wait_fixed(3)):
+                with attempt:
+                    r = requests.get(
+                        f"{'https' if self._tls_enabled else 'http'}://{self._primary_endpoint}:8008/health",
+                        verify=self._verify,
+                    )
+        except RetryError:
+            return False
+
+        return r.json()["state"] == "running"
```

Comment:
If the state isn't running is it worth retrying? ie
```
        try:
            for attempt in Retrying(stop=stop_after_delay(10), wait=wait_fixed(3)):
                with attempt:
                    r = requests.get(
                        f"{'https' if self._tls_enabled else 'http'}://{self._primary_endpoint}:8008/health",
                        verify=self._verify,
                    )
                    if not r.json()["state"] == "running":
                         raise endpointNotReadyError
        except RetryError: 
            return False

        return True
```

Reply:
Yes. It is a great improvement! I will implement it. Thanks Mia!

Reply:
Updated on https://github.com/canonical/postgresql-k8s-operator/pull/46/commits/fbb43329a331d55cc2aee2f304d2f5adf8bec838.